jme3-core/src/main/java/com/jme3/material/TechniqueDef.java

```diff
@@ -715,6 +816,46 @@ public void read(JmeImporter im) throws IOException{
         usesNodes = ic.readBoolean("usesNodes", false);
         shaderNodes = ic.readSavableArrayList("shaderNodes", null);
         shaderGenerationInfo = (ShaderGenerationInfo) ic.readSavable("shaderGenerationInfo", null);
+
+        final int worldBindOrdinals[] = ic.readIntArray("worldBinds", EMPTY_INT_ARRAY);
+
+        worldBinds = new ArrayList<>(worldBindOrdinals.length);
+
+        for (final int ordinal : worldBindOrdinals) {
+            worldBinds.add(UniformBinding.valueOf(ordinal));
```

Comment:
@Nehon it's an example of using Enum.valueOf(index);

Reply:
Well if it's for serialization then it's not a good example. Doing it right would be to add the corresponding method in outputcapsule to save a list of Enum, and to save the values.
Saving the ordinal is very bad practice as if the order changes one day into the enum it would break the old models loading. We already have this in other places in the engine and it's very bad. Let's not do the same mistakes over and over again.
Anyway. This PR won't make it to core.

Reply:
@Nehon I can remake it to use transform from enum list to string list...